app/models/concerns/user_omniauth_concern.rb

```diff
@@ -0,0 +1,34 @@
+module UserOmniauthConcern
+  extend ActiveSupport::Concern
+  include UserOmniauthFacebookConcern
+
+  included do
+    devise :omniauthable
```

Comment:
No need, For this I debugged into devise's source code, they can automatically know the providers from the configuration in devise initialisers.